cli/popper/runner_slurm.py

```diff
@@ -151,3 +152,72 @@ def _create_cmd(self, step, img, cid):
         cmd.append(f'{image} {command}')
 
         return ' '.join(cmd)
+
+
+class SingularityRunner(SlurmRunner, HostSingularityRunner):
+
+    def __init__(self, **kw):
+        super(SingularityRunner, self).__init__(**kw)
+        if self._config.reuse:
+            log.fail('Reuse not supported for SingularityRunner.')
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        pass
+
+    def run(self, step):
+        self._setup_singularity_cache()
+        cid = pu.sanitized_name(step['name'], self._config.wid) + '.sif'
+        self._container = os.path.join(self._singularity_cache, cid)
+
+        build, img, build_source = self._get_build_info(step)
+
+        cmd = []
+        cmd.append(f'rm -rf {self._container}')
```

Comment:
In an HPC cluster, one usually has an NFS mounted that is shared by all the nodes of SLURM jobs. Most of the time (if not always), this is the `$HOME` folder. Thus, in the case of singularity, since the image that Popper prepares is stored in the cache folder (which in turn is on `HOME`), we can execute the commands that remove, build, fetch, etc. images outside of a job SLURM. 

So we can run the `rm -r <container>` and `singularity build/pull` commands using `HostHostRunner._exec_cmd`. Then we can add only the `singularity exec/run` in a SLURM batch job.

Reply:
@ivotron Should we also try to remove the build/pull steps from slurm job in docker?

Reply:
for docker is different because the cache is not usually on the home folder, so every node needs to remove, pull/build, etc. So we can leave Docker as is.

Reply:
cool.